### ISSUE TITLE
Crash when user / role is removed

### ISSUE THREAD

**Issue by andresriancho** [nccgroup/PMapper#11]
**Describe the bug**
`pmapper.py graph` will crash when a user / role is removed during the pull process:

```
[andres:~/tools/PMapper-fork] [venv] master(+2/-2)* 255 ± python pmapper.py graph
Using profile: default
Pulling data for account 080387373610
Using principal with ARN arn:aws:iam::080387373610:user/infosec..vpcreadonly
[+] Pulling info on IAM users and roles, finding admins.
Principals Checked: 100%|################################################################################################################################################| 6641/6641 [57:31<00:00,  2.41it/s]
[+] Finished finding admins.
[+] Started EC2 checks.
Principals Checked:  30%|###########################################4                                                                                                    | 2006/6641 [32:26<45:42,  1.69it/s]Traceback (most recent call last):
  File "pmapper.py", line 176, in <module>
    main()
  File "pmapper.py", line 50, in main
    handle_graph(parsed)
  File "pmapper.py", line 59, in handle_graph
    graph = pull_graph(parsed.profile)
  File "pmapper.py", line 136, in pull_graph
    enumerator.fillOutGraph()
  File "/home/eth/tools/PMapper-fork/principalmap/enumerator.py", line 32, in fillOutGraph
    checkrunner.runChecks()
  File "/home/eth/tools/PMapper-fork/principalmap/edgeconditions/checkrunner.py", line 50, in runChecks
    edgelist = checker.performChecks(self.session, self.graph.nodes)
  File "/home/eth/tools/PMapper-fork/principalmap/edgeconditions/ec2checks.py", line 56, in performChecks
    ActionNames=['ec2:RunInstances', 'ec2:AssociateIamInstanceProfile', 'iam:CreateInstanceProfile'],
  File "/home/eth/tools/PMapper-fork/venv/local/lib/python2.7/site-packages/botocore/client.py", line 320, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/eth/tools/PMapper-fork/venv/local/lib/python2.7/site-packages/botocore/client.py", line 623, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.NoSuchEntityException: An error occurred (NoSuchEntity) when calling the SimulatePrincipalPolicy operation: Unable to retrieve specified IAM entity.
Exception KeyError: KeyError(<weakref at 0x7f01e803e890; to 'tqdm' at 0x7f01e9f1ddd0>,) in <bound method tqdm.__del__ of Principals Checked:  30%|###########################################4                                                                                                    | 2006/6641 [32:26<45:42,  1.69it/s]> ignored
[andres:~/tools/PMapper-fork] [venv] master(+2/-2)* 1h31m4s 1 ± 
```

**To Reproduce**
Just run `pmapper.py graph` on an AWS organization where things change: user / role is removed.

**Expected behavior**
Ignore the user that was deleted and continue with the next.

**Comment by andresriancho** [comment]
Wrote fix for this issue. Testing now.

**Comment by raviadds** [comment]
was this ever merged bruv

**Comment by ncc-erik-steringer** [comment]
@raviadds , haven't seen a pull request. This issue seems to have a lot of interest, so I can prioritize it over the next few weeks.

**Comment by ncc-erik-steringer** [comment]
This issue should be mostly fixed with the release of v1.0.0: https://github.com/nccgroup/PMapper/releases/tag/v1.0.0 

This might not be something that can be completely fixed, since PMapper is basically trying to take a snapshot of the account and work on top of that. If things are changing underneath it, that's like hitting a moving target.

Closing.
